src/app/pages/questions/containers/questions-page.component.ts

```diff
@@ -206,15 +206,16 @@ export class QuestionsPageComponent implements OnInit {
   }
 
   nextAction(event) {
-    if (event == NextButtonEventType.AUTO) return this.nextQuestion()
+    if (event == NextButtonEventType.AUTO)
+      return setTimeout(() => this.nextQuestion(true), 100)
     if (event == NextButtonEventType.ENABLE)
       return setTimeout(() => this.updateToolbarButtons(), 100)
     if (event == NextButtonEventType.DISABLE)
       return (this.isRightButtonDisabled = true)
   }
 
-  nextQuestion() {
-    if (this.isRightButtonDisabled) return
+  nextQuestion(force?: boolean) {
+    if (this.isRightButtonDisabled && !force) return
```

Comment:
can you explain why this is needed? From what i understand the if/else above, next will only be triggered when either the question is auto type or when the next button is pressed (which will be disabled until user selects an answer in case of non-auto types)? why is this extra check and force necessary?

Reply:
Ah yes I've removed it now. This is not needed anymore since the right button disabling is already done in the UI, we don't need the additional check here. I refactored it before but forgot to remove this bit.